lib/vault/client.rb

```diff
@@ -177,8 +179,20 @@ def request(verb, path, data = {}, headers = {})
       # Create the HTTP connection object - since the proxy information defaults
       # to +nil+, we can just pass it to the initializer method instead of doing
       # crazy strange conditionals.
-      connection = Net::HTTP.new(uri.host, uri.port,
-        proxy_address, proxy_port, proxy_username, proxy_password)
+      connection = Net::HTTP::Persistent.new(name: "vault-ruby")
+
+      if proxy_address
+        proxy_uri = URI.parse "http://#{proxy_address}"
```

Comment:
I don't know where the comment went, but doesn't this eliminate the ability for someone to use an https proxy?